command/deployment_status.go

```diff
@@ -208,7 +208,7 @@ func isStdoutTerminal() bool {
 // but only used for tty and non-Windows machines since glint doesn't work with
 // cmd/PowerShell and non-interactive interfaces
 // Margins are used to match the text alignment from job run
-func (c *DeploymentStatusCommand) ttyMonitor(client *api.Client, deployID string, index uint64, verbose bool) {
+func (c *DeploymentStatusCommand) ttyMonitor(client *api.Client, deployID string, index uint64, verbose bool) (status string, err error) {
```

Comment:
There's an "inner" `c.ttyMonitor` call that happens on rollback. 

When the outer function returns, we're swallowing the status for the rollback deployment, which seems correct. But we're also swallowing any error returned by the inner `c.ttyMonitor`. If the outer monitor function's `err` is nil, then we should probably set `err` to the error returned by the inner `c.ttyMonitor`. There's no current case where `(*monitor).monitor()` does anything that'd be useful with that error, but it's a place that's easy for someone to change later and then we'd miss the inner error. It's also potentially handy for tests.

(This also applies to `c.defaultMonitor` and its inner `c.defaultMonitor`.)

Reply:
I had decided to swallow that inner error because I figured that the most important information to a caller will be whether the deployment succeeded. As far as I understand Go though a caller wouldn't usually use the value of `status` if `err != nil` because a function's return value is considered unreliable if the function errored? So we'd lose that information. Sorry if this is a beginner's question, I'm still new to the language.

Reply:
Hm... you've got a point there. "Ideomatic go" would say the first value in the returned tuple should be the empty value, but in this case we're using it because it's a status string. Tell you what, let's leave it as you have it here; I'm thinking of future code that hasn't been written yet and might never be written. YAGNI 😀